lib/Runtime/ByteCode/ByteCodeEmitter.cpp

```diff
@@ -10656,6 +10749,16 @@ void Emit(ParseNode *pnode, ByteCodeGenerator *byteCodeGenerator, FuncInfo *func
         ParseNode *lhs = pnode->AsParseNodeBin()->pnode1;
         ParseNode *rhs = pnode->AsParseNodeBin()->pnode2;
         byteCodeGenerator->StartStatement(pnode);
+
```

Comment:
Is this condition trying to detect whether the op is going to write to closure rather than to a register? Is that the purpose of the check on the funcInfo?

Reply:
The API called Symbol::IsInSlot is better suited to that, in general. It tracks things like whether we're currently storing a symbol to its slot or are keeping it in register because we know that no nested function needs to read it yet.